tests/thread_singleton_test.py

```diff
@@ -0,0 +1,73 @@
+import dace
+from dace import dtypes, nodes
+import numpy as np
+
+
+N = dace.symbol("N")
+
+
+@dace.program
+def cpu_singleton_program(A: dace.int32[N]):
+    for i in dace.map[0:N]:
+        A[i] = i
+
+
+@dace.program
+def cpu_singleton_program_invalid_map_ranges(A: dace.int32[N]):
+    # We can only have one range with the `CPU_Multicore_Singleton` schedule!
+    for i, j in dace.map[0:N, 0:N]:
+        A[i] = i
+
+
+def first_map_entry_to_cpu_singleton(sdfg: dace.SDFG) -> None:
+    map_entry = None
+    for node, _ in sdfg.all_nodes_recursive():
+        if isinstance(node, nodes.MapEntry):
+            map_entry = node
+            break
+
+    map_entry.schedule = dtypes.ScheduleType.CPU_Multicore_Singleton
+
+
+def test_cpu_singleton() -> None:
+    N = 128
```

Comment:
This test will become useless if OpenMP uses 128 or more threads. We should probably protect against that.